app/Models/EntityEvent.php

```diff
@@ -191,16 +191,38 @@ public function scopeCalendarDate(Builder $query)
      */
     public function readableDate(): string
     {
+
         if ($this->readableDate === null) {
             // Replace month with real month, and year maybe
             $months = $this->calendar->months();
             $years = $this->calendar->years();
 
             try {
-                $this->readableDate = $this->day . ' ' .
-                    (isset($months[$this->month - 1]) ? $months[$this->month - 1]['name'] : $this->month) . ', ' .
-                    ($years[$this->year] ?? $this->year) . ' ' .
-                    $this->calendar->suffix;
+                if ($this->calendar->format) {
+                    $customFormat = '';
+
+                    foreach (mb_str_split($this->calendar->format) as $char) {
```

Comment:
You can use Str::replace instead of a complicate if-else
```php
$customFormat = Str::replace(
  ['d', 's', 'y', 'm', 'M'],
  [$this->day, $this->calendar->suffix, $years[$this->year] ?? $this->year, $this->month, isset($months[$this->month - 1]) ? $months[$this->month - 1]['name'] : $this->month)],
  $customFormat
);
```